draft-ietf-quic-transport.md

```diff
@@ -1548,41 +1548,56 @@ The fields of a CONNECTION_CLOSE frame are as follows:
 * Reason Phrase: An optional human-readable explanation for why the connection
   was closed.
 
+
 ## GOAWAY Frame {#frame-goaway}
 
-An endpoint may use a GOAWAY frame (type=0x03) to notify its peer that the
-connection should stop being used, and will likely be closed in the future. The
-endpoints will continue using any active streams, but the sender of the GOAWAY
-will not initiate any additional streams, and will not accept any new streams.
-The frame is as follows:
+An endpoint uses a GOAWAY frame (type=0x03) to initiate a graceful shutdown of a
+connection.  The endpoints will continue to use any active streams, but the
+sender of the GOAWAY will not initiate or accept any additional streams beyond
+those indicated.  The GOAWAY frame is as follows:
 
 ~~~
  0                   1                   2                   3
  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                        Error Code (32)                        |
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                     Last Good Stream ID (32)                  |
+|                    Last Client Stream ID (32)                 |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|   Reason Phrase Length (16)   |      [Reason Phrase (*)]    ...
+|                    Last Server Stream ID (32)                 |
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 ~~~
 
-The fields of a GOAWAY frame are as follows:
+The fields of a GOAWAY frame are:
 
-* Error Code: A 32-bit field error code which indicates the reason for closing
-  this connection.
+Last Client Stream ID:
 
-* Last Good Stream ID: The last Stream ID which was accepted by the sender of
-  the GOAWAY message.  If no streams were replied to, this value must be set to
-  0.
+: The last client-initiated Stream ID which was accepted by the sender of the
+  GOAWAY frame.  All higher-numbered, client-initiated streams (that is, odd
+  numbered streams) are implicitly reset by sending or receiving the GOAWAY
+  frame.
 
-* Reason Phrase Length: A 16-bit unsigned number specifying the length of the
-  reason phrase.  This may be zero if the sender chooses to not give details
-  beyond the error code.
+Last Server Stream ID:
+
+: The last server-initiated Stream ID which was accepted by the sender of the
```

Comment:
Because the sender isn't necessarily the server, and the client isn't "creating" server-initiated streams.  Maybe we could say "the highest ... ID on which the sender has processed or will process data"?  But that doesn't seem more compact, so I'm not sure it's an improvement.

But regardless, that's feedback on #354, not this change in particular.